src/components/Search.js

```diff
@@ -1,34 +1,116 @@
+    SEARCH_GENRE
+} from '../constants';
+
+
+class Search extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        const searchParams = new URLSearchParams(this.props.location.search);
+
+        this.state = {
+            searchParams,
+            searchString: searchParams.get(SEARCH_PARAM_NAME) || '',
+            searchBy: searchParams.get(SEARCHBY_PARAM_NAME) || SEARCH_TITLE
+        };
+    }
+
+    handleChange = (event) => {
+        this.setState({
+            ...this.state,
+            searchString: event.target.value || '',
+        });
+    };
+
+    setSearchType = (type) => {
+        this.setState({
+            ...this.state,
+            searchBy: type
+        });
+    };
+
+    submit = (event) => {
+        this.state.searchParams.set(SEARCHBY_PARAM_NAME, this.state.searchBy);
+        this.state.searchParams.set(SEARCH_PARAM_NAME, this.state.searchString);
+
+        const search = this.state.searchParams.toString();
+
+        this.props.history.push({
+            ...this.props.location,
+            search
+        });
+
+        this.props.fetchData(search);
+
+        event.preventDefault();
+    };
 
     render() {
+        const _this = this;
+
         return (
             <React.Fragment>
                 <div className="form-row mt-3 mb-3">
-                    <div className="col">
+                    <form className="col" onSubmit={this.submit}>
                         <label htmlFor="inputSearch">
                             <span className="text-light">FIND YOUR MOVIE</span>
                         </label>
-                        <input type="text" className="form-control" id="inputSearch"
-                               placeholder="Input your request"/>
-                    </div>
+                        <input id="inputSearch"
+                               type="text"
+                               className="form-control"
+                               placeholder="Input your request"
+                               value={this.state.searchString}
+                               onChange={this.handleChange}
+                        />
+                    </form>
                 </div>
                 <div className="form-row pb-4">
                     <div className="col">
                         <label className="mr-2">
                             <small className="text-light">SEARCH BY</small>
                         </label>
-                        <Button title="TITLE" isActive={true} size="small" addClasses="mr-1"/>
-                        <Button title="GENRE" size="small" type="secondary"/>
+                        <Button title="TITLE"
+                                isActive={this.state.searchBy === SEARCH_TITLE}
+                                size="small"
+                                addClasses="mr-1"
+                                type="secondary"
+                                onClick={() => _this.setSearchType(SEARCH_TITLE)}
```

Comment:
There is no need for _this trick here. You can use .this syntax, as you did with this.submit

Reply:
babel throws error if `this` context uses, looks like babel issue:
<img width="1203" alt="2018-05-30 00 14 14" src="https://user-images.githubusercontent.com/16851711/40686857-6308763e-63a1-11e8-8971-d7268901bf8e.png">